### ISSUE TITLE
How to use legacy version for maximum compatibility?

### ISSUE THREAD

**Issue by bluepuma77** [nimiq/qr-scanner#143]
For maximum compatibility with older mobile devices I would like to use the legacy version.

I would have expected to use it this way:
```
<script src="path/to/qr-scanner.legacy.min.js"></script>
<script>
    // do something with QrScanner
</script>

```
But I see errors in the browser console and `QrScanner` is undefined after the script is loaded:

```
Uncaught SyntaxError: export declarations may only appear at top level of a module [qr-scanner.legacy.min.js:53:46]
Uncaught ReferenceError: QrScanner is not defined
```

**Comment by danimoh** [comment]
Hello @bluepuma77.
The legacy build is an es6 module which can be used in other scripts of `type` `module`, for example as follows:

```js
<script type="module">
    import QrScanner from 'path/to/qr-scanner.legacy.min.js';
    // do something with QrScanner
</script>
```

I just saw however, that support for es modules is actually not that great for some of the old browsers that already supported es6.
In the next release I'll therefore switch to building the legacy build as a umd (universal module definition) bundle which will then be able to be used as you expected without an es module import.

FYI, the regular `qr-scanner.umd.min.js` umd build already has great browser support.
The usage of [dynamic imports](https://caniuse.com/es6-module-dynamic-import) and [async functions](https://caniuse.com/async-functions) in the regular build are probably the features that limit browser support the most, but that should already get us to ~94% user support.
The legacy build as an umd could maybe [get the support up to ~96-98%](https://caniuse.com/es6).
But if you want to use web cam scanning and not just image scanning, you'll be limited to browsers that support [getUserMedia](https://caniuse.com/mdn-api_mediadevices_getusermedia) anyways, which is also ~94%.
For now I would recommend you to use the regular `qr-scanner.umd.min.js` and switch to the legacy umd build on next release only if you don't do camera scanning.

As the browser support is already pretty good without the legacy build, I also plan to only provide it until the next major release.
It was just added, when I introduced async functions and dynamic imports into the code recently, to not break support for existing users until the next major release.
